selectors/selector-syntax.md

```diff
@@ -0,0 +1,430 @@
+
+#### URL Friendly
+
+Selector syntax should embed easily inside URLs.
+
+This means where possible, this syntax restricts itself to the characters that can be embedded in URLs without needing to escape them. This means this subset of ASCII:
+
+```js
+[ '!', "'", '(', ')', '*', '-', '.', '0', '1',
+  '2', '3', '4', '5', '6', '7', '8', '9', 'A',
+  'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J',
+  'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S',
+  'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '_', 'a',
+  'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
+  'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's',
+  't', 'u', 'v', 'w', 'x', 'y', 'z', '~']
+```
+
+This also also means it needs to be as terse as possible and not contain whitespace of any kind.
+
+For example, this selector simulates a git shallow clone by recursively walking commit parents up to depth 5 and walking all of the tree graphs for each.
+
+```ipldsel
+# Starting at the commit block.
+R5f'tree'R*~'parents'*~
+```
+
+#### Human Friendly
+
+Selector syntax should be easy to read/author by humans.
+
+This means it should be terser than the JSON or YAML representations of the IPLD data, but still verbose enough to have meaningful structure and keywords/symbols.
+
+This means it should allow flexibility with whitespace as well as allowing optional symbols and annotations to make structure easier to see visually.
+
+The exact same selector for git shallow clone from above can also be written in the following style: (This is not another mode, it's the same syntax):
+
+```ipldsel
+recursive(limit=5
+  fields(
+    'tree'(
+      recursive(
+        all(recurse)
+      )
+    )
+    'parents'(
+      all(recurse)
+    )
+  )
+)
+```
+
+Examples
+--------
+
+### Deeply Nested Path
+
+Based on [this example](https://github.com/creationix/specs/blob/selector_text_syntax/selectors/example-selectors.md#deeply-nested-path).
+
+A selector to extract the year:
+
+#### Human Readable Style
+
+This is the default style for human interfacing.  It has clear structure and descriptive keywords.
+
+```ipldsel
+fields('characters'(
+  fields('kathryn-janeway'(
+    fields('birthday'(
+      fields('year'(match))
+    ))
+  ))
+))
+```
+
+#### URL Embeddable Style
+
+This is the default style for maximum terseness.  It minifies everything possible.
+
+```ipldsel
+f'characters'f'kathryn-janeway'f'birthday'f'year'.
+```
+
+#### Using multi-segment-path extension.
+
+We can add some syntax sugar extensions as macros for common use cases as they come up.
+
+```ipldsel
+p'characters/kathryn-janeway/birthday/year'.
+```
+
+This example is sugar for a deeply nested path using a new `"paths"`/`"p"` keyword.  It's a version of `"f"` that splits its input on `/` and creates nested selectors.
+
+This can be done as a macro-like sugar at this level.  Another option is for the selector spec to add `ExplorePaths` as a new selector with optimized representation and execution.
```

Comment:
> "Another option is for ..."

I'm not sure that it's appropriate to put enhancement suggestions inline in this way; implementers and users reading this aren't going to find this helpful. So maybe find a way to split it out into a separate doc detailing possible future enhancements or just file an issue or even open a PR to add the change if you think it's important enough?

Reply:
Agreed, I'll remove it.  In another conversation we discovered that the current short form without the enhancement is just as terse and doesn't depend on semi-url-unsafe characters like `/`.